Rubberduck.Core/AutoComplete/AutoCompleteBase.cs

```diff
@@ -10,27 +13,29 @@ protected AutoCompleteBase(string inputToken, string outputToken)
             OutputToken = outputToken;
         }
 
+        public bool IsInlineCharCompletion => InputToken.Length == 1 && OutputToken.Length == 1;
         public bool IsEnabled { get; set; }
         public string InputToken { get; }
         public string OutputToken { get; }
 
-        public virtual bool Execute(AutoCompleteEventArgs e)
+        public virtual bool Execute(AutoCompleteEventArgs e, AutoCompleteSettings settings)
```

Comment:
Because I didn't want to accidentally cache settings, nor accidentally load fresh settings at every keypress... Injecting it here puts that responsibility on the caller and out of AC code.